app/decorators/service_provider_session_decorator.rb

```diff
@@ -48,4 +52,18 @@ def sp_return_url
   private
 
   attr_reader :sp, :view_context
+
+  def sp_logo_file_exist?
```

Comment:
wondering if this will be an expensive query to run on page load. And if so, should we just let it be a broken image if it is not there?